packages/bootstrap-chat/src/composed/ChatApp.jsx

```diff
@@ -75,39 +78,55 @@ function ChatApp({
   }
 
   function trackAgentName(agentName) {
-    if (agentName) {
-      const previouslyTrackedNames = taskContext.currentAgentNames || {};
-      const newAgentName = { [agentId]: agentName };
-      const currentAgentNames = { ...previouslyTrackedNames, ...newAgentName };
-      updateContext({ currentAgentNames: currentAgentNames });
-    }
+    const previouslyTrackedNames = taskContext.currentAgentNames || {};
+    const newAgentName = { [agentId]: agentName, agentName: agentName };
```

Comment:
what is the purpose of `agentName` here? currently it will be overwritten by each subsequent agent that is tracked, since they are spread onto `currentAgentNames`. something seems off here

Reply:
This ends up covering both current and previous names that we should consider to be "isSelf", but it also has a mapping for us to represent incoming messages with their correct ids. My intent was to have, for instance: 
```
{
   onboarding_1: "OnboardingAgent 1",
   "OnboardingAgent 1": "OnboardingAgent 1",
   task_agent_1: "In-task Agent",
   "In-task Agent": "In-task Agent",
}
```
be constructed by an agent first receiving `onboarding_1: "OnboardingAgent 1",` and later getting `task_agent_1: "In-task Agent",`. Then when we get incoming messages either with sender id `onboarding_1` or `OnboardingAgent 1` we know it was actually from ourself. The cause for having both allows us to get this detail on reconnect, as `ParlaIChatBlueprint` doesn't retain the mephisto-based sender ID in the raw messages later sent on reconnect.

Reply:
Perhaps you meant to have brackets around `[agentName]` as it is now, your object would look like this:

```
{
   onboarding_1: "OnboardingAgent 1",
   task_agent_1: "In-task Agent",
   agentName: "In-task Agent",
}

```

Reply:
Indeed - good catch!